pkg/queue/manager.go

```diff
@@ -425,23 +441,27 @@ func (m *Manager) QueueInadmissibleWorkloads(ctx context.Context, cqNames sets.S
 	}
 }
 
-// queueAllInadmissibleWorkloadsInCohort moves all workloads in the same
+// moveWorkloadsCq moves all workloads in the same
 // cohort with this ClusterQueue from inadmissibleWorkloads to heap. If the
 // cohort of this ClusterQueue is empty, it just moves all workloads in this
 // ClusterQueue. If at least one workload is moved, returns true, otherwise
 // returns false.
 // The events listed below could make workloads in the same cohort admissible.
-// Then queueAllInadmissibleWorkloadsInCohort need to be invoked.
+// Then moveWorkloadsCq need to be invoked.
 // 1. delete events for any admitted workload in the cohort.
 // 2. add events of any cluster queue in the cohort.
 // 3. update events of any cluster queue in the cohort.
-func (m *Manager) queueAllInadmissibleWorkloadsInCohort(ctx context.Context, cq *ClusterQueue) bool {
-	if !cq.HasParent() {
-		return cq.QueueInadmissibleWorkloads(ctx, m.client)
+// 4. update of cohort.
+func (m *Manager) moveWorkloadsCq(ctx context.Context, cq *ClusterQueue) bool {
+	if cq.HasParent() {
+		return m.moveWorkloadsCohort(ctx, cq.Parent())
 	}
+	return cq.QueueInadmissibleWorkloads(ctx, m.client)
+}
 
+func (m *Manager) moveWorkloadsCohort(ctx context.Context, cohort *cohort) bool {
```

Comment:
Wouldn't we have to requeue all workloads in the entire tree?

Reply:
Yes, but I will do this in a follow-up PR "Support Hierarchical Cohorts in Queue package" - similar to #3006. We may reuse cycle-detection logic from that PR